### ISSUE TITLE
Cannot start the framework

### ISSUE THREAD

**Issue by veli4ko91** [eclipse/concierge#73]
Hey, I probably have missed something obvious, but what am I doing wrong here?

`D:\_osgi\concierge-incubation-6.0.0.SNAPSHOT.tar\concierge-incubation-6.0.0.SNAPSHOT\concierge-incubation-6.0.0.SNAPSHOT>java -jar framework/org.eclipse.concierge-6.0.0.SNAPSHOT.jar samples/default.xargs

Exception in thread "main" org.osgi.framework.BundleException: Illegal attribute name ";java.vm.version="25.131-b11";org.osgi.supports.framework.fragment="true";user.variant="";java.awt.printerjob="sun.awt.windows.WPrinterJob";sun.io.unicode.encoding="UnicodeLittle";org.osgi.framework.version="1.5.0";org.osgi.framework.storage="storage";awt.toolkit="sun.awt.windows.WToolkit";sun.stdout.encoding="cp437";org.osgi.framework.bundle.parent="boot";user.script="";org.osgi.supports.bootclasspath.extension="false";user.home="C:\Users\valkov";java.specification.vendor="Oracle Corporation";java.library.path="C:\jdk1.8.0_131c1\bin
        at org.eclipse.concierge.Utils.parseLiterals(Utils.java:162)
        at org.eclipse.concierge.Resources$GenericReqCap.<init>(Resources.java:74)
        at org.eclipse.concierge.Resources$BundleCapabilityImpl.<init>(Resources.java:138)
        at org.eclipse.concierge.Concierge.init(Concierge.java:1316)
        at org.eclipse.concierge.Concierge.init(Concierge.java:1120)
        at org.eclipse.concierge.compat.service.XargsFileLauncher.processXargsInputStream(XargsFileLauncher.java:94)
        at org.eclipse.concierge.compat.service.XargsFileLauncher.processXargsFile(XargsFileLauncher.java:83)
        at org.eclipse.concierge.Concierge.doMain(Concierge.java:617)
        at org.eclipse.concierge.Concierge.main(Concierge.java:536)
`

It seems that something is wrong with my passing of properties.
Thanks!

**Comment by rellermeyer** [comment]
Just guessing but based on the output: could it be that you have a dangling quotation mark in your xargs file?

**Comment by veli4ko91** [comment]
Doesn't seem like it. I used the default one:

`-Dorg.osgi.framework.storage.clean=onFirstInit
-istart bundles/org.eclipse.concierge.shell-5.0.0.*.jar`

**Comment by tmarkwardt** [comment]
hi,
with r6 branch i have this error also:

`P:\devel\Eclipse\Concierge3>p:\progs\java\jdk1.8.0_161.w64\bin\java -jar "framework\org.eclipse.concierge-6.0.0.SNAPSHOT-20190606113805.jar" samples\default.xargs
Exception in thread "main" org.osgi.framework.BundleException: Illegal attribute name ";java.vm.version="25.161-b12";org.osgi.supports.framework.fragment="true";user.variant="";java.awt.printerjob="sun.awt.windows.WPrinterJob";sun.io.unicode.encoding="UnicodeLittle";org.osgi.framework.version="1.5.0";org.osgi.framework.storage="storage";awt.toolkit="sun.awt.windows.WToolkit";sun.stdout.encoding="cp850";org.osgi.framework.bundle.parent="boot";user.script="";org.osgi.supports.bootclasspath.extension="false";user.home="C:\Users\xxxxxx";java.specification.vendor="Oracle Corporation";java.library.path="p:\progs\java\jdk1.8.0_161.w64\bin
        at org.eclipse.concierge.Utils.parseLiterals(Utils.java:162)
        at org.eclipse.concierge.Resources$GenericReqCap.<init>(Resources.java:74)
        at org.eclipse.concierge.Resources$BundleCapabilityImpl.<init>(Resources.java:138)
        at org.eclipse.concierge.Concierge.init(Concierge.java:1286)
        at org.eclipse.concierge.Concierge.init(Concierge.java:1096)
        at org.eclipse.concierge.compat.service.XargsFileLauncher.processXargsInputStream(XargsFileLauncher.java:94)
        at org.eclipse.concierge.compat.service.XargsFileLauncher.processXargsFile(XargsFileLauncher.java:83)
        at org.eclipse.concierge.Concierge.doMain(Concierge.java:617)
        at org.eclipse.concierge.Concierge.main(Concierge.java:536)`

**Comment by tmarkwardt** [comment]
and here a fix variant ...

`--- a/framework/org.eclipse.concierge/src/org/eclipse/concierge/Concierge.java
+++ b/framework/org.eclipse.concierge/src/org/eclipse/concierge/Concierge.java
@@ -1280,7 +1280,7 @@ public final class Concierge extends AbstractBundle implements Framework,
                        String key = en.nextElement().toString();
                        if(key.startsWith("osgi.native"))
                                continue;
-                       nativeCapBuilder.append(key).append("=\"").append(properties.getProperty(key).toString()).append("\";");
+                       nativeCapBuilder.append(key).append("=\"").append(properties.getProperty(key).toString().replace( "\\", "\\\\").replace( "\"", "\\\"")).append("\";");
                }
                final BundleCapabilityImpl nativeCap = new BundleCapabilityImpl(this,
                                nativeCapBuilder.toString());`

the problem is tha java property file.path on Windows. the value is a backslash. on merge and split of strings, this value mask a finish Quote and breaked the system property working.

Regards,
Torsten

**Comment by tverbele** [comment]
Thanks!

**Comment by falmanna** [comment]
Hi @tmarkwardt 
Are these PRs available in the latest [SNAPSHOT](https://download.eclipse.org/concierge/download/snapshots/?d) build? 

I am getting the same errors using `concierge-incubation-SNAPSHOT-latest.zip` build.

**Comment by tmarkwardt** [comment]
Hi @falmanna 
sorry but my changes was from june and the snaphot was from may ... please download and translate from source here ...
Regards

**Comment by falmanna** [comment]
Thanks @tmarkwardt 
Actually, I ran the 6.0-SNAPSHOT on a Linux system and it passed the problem, but it seems something is missing. All bundles are showing active in the CLI, but they are not responding. 

Cloud the problem be caused by the fact that some of the installed bundles depend on OSGi R6?  I am assuming that the latest snapshot does support this, doesn't it?

**Comment by tmarkwardt** [comment]
i can't found a problem with https://github.com/eclipse/concierge.git to build or start. i have tested under debian stretch now. only the sample files for arguments is inconsistent - the start level bundle is 5.0.0 but the sample used 5.1.0

**Comment by falmanna** [comment]
@tmarkwardt 
I didn't try to build this repo myself, I used the pre-built SNAPSHOT I mentioned before. 

What I am trying to accomplish is, I need to start some bundles that have depends on OSGi R6, and according to the website concierge version `5.1.0`  only support R5, so I tried to use version `6.0.0-SNAPSHOT` that should support R6 annotation according to [this](https://github.com/eclipse/concierge/pull/58) pull request.

After running the snapshot, everything seemed successful, but I found that none of the bindings is responding as if they didn't register their services correctly.

I think I might wait until the stable release of version `6.0.0` (which I hope will be soon) :)
